monai/transforms/intensity/array.py

```diff
@@ -1137,3 +1130,120 @@ def __call__(self, img: np.ndarray) -> np.ndarray:
         return np.asarray(
             np.interp(img, reference_control_points_scaled, floating_control_points_scaled), dtype=img.dtype
         )
+
+
+class RandGibbsNoise(RandomizableTransform):
+    """
+    Naturalistic image augmentation via Gibbs artifacts. The transform 
+    randomly applies Gibbs noise to 3D MRI images. Gibbs artifacts
+    are one of the common type of type artifacts appearing in MRI scans.
+    
+    For general information on Gibbs artifacts, please refer to:
+    https://pubs.rsna.org/doi/full/10.1148/rg.313105115
+    https://pubs.rsna.org/doi/full/10.1148/radiographics.22.4.g02jl14949
+
+
+    Args:
+        keys: 'image', 'label', or ['image', 'label'] depending on which data
+                you need to transform.
+        alpha: Parametrizes the intensity of the Gibbs noise filter applied. Takes
+            values in [0,1] with alpha = 0 acting as the identity mapping. 
+            If a length-2 list is given as [a,b] then the value of alpha will be 
+            sampled uniformly from the interval [a,b]. 0 <= a <= b <= 1.
+        prob: probability of applying the transform. 
+
+    """
+
+    def __init__(self, alpha: Union[float, List[float]] = 0.5, prob: float = 0.1,) -> None:
+
+        assert prob <= 1 and prob >= 0, "prob must take values in [0,1]."
```

Comment:
delete this since you have the if/else statement

Reply:
Ok, will remove. The __init__ of the `RandomizableTransform ` superclass also has the line`self.prob = min(max(prob, 0.0), 1.0)` which caps any erroneous values of `prob`. My reasoning was to alert the user that he/she had a typo inputting the probability value as oppose to just letting that typo be mapped to 0 or 1 by the min-max.

Reply:
Sorry I realise my comment didn't make sense, I mistook `prob` for `alpha` so thought it was code duplication. Anyway, as you say, the `RandomizableTransform` base class should take care of that.